### PR TITLE
feat(outbound-app): add user token management functions

### PR DESCRIPTION
**Pull request by omercnet**
Fixes descope/etc#13474

[View workflow run](https://github.com/descope/shuni/actions/runs/20659261150)

Perfect! I have successfully added support for Outbound App management functions to the Go SDK. Here's a summary of what was implemented:

## Summary

I've successfully added three new Outbound App management functions to the Go SDK:

### 1. **FetchOutboundAppUserToken**
   - Fetches an outbound application user token with specified scopes
   - Endpoint: `POST /v1/mgmt/outbound/app/user/token`
   - Supports optional configurations like `withRefreshToken` and `forceRefresh`

### 2. **DeleteOutboundAppUserTokens**
   - Deletes outbound app user tokens by appID or userID (at least one required)
   - Endpoint: `DELETE /v1/mgmt/outbound/user/tokens`
   - Flexible filtering by app or user

### 3. **DeleteOutboundAppTokenByID**
   - Deletes a specific outbound app token by its ID
   - Endpoint: `DELETE /v1/mgmt/outbound/token`

### Changes Made:

1. **API Routes** (`descope/api/client.go`):
   - Added route definitions for all three new endpoints

2. **Type Definitions** (`descope/types.go`):
   - `FetchOutboundAppUserTokenRequest` - Request structure
   - `OutboundAppUserTokenOptions` - Options for token fetching
   - `OutboundAppUserToken` - Token response structure
   - `FetchOutboundAppUserTokenResponse` - Wrapper for token response

3. **Implementation** (`descope/internal/mgmt/outbound_application.go`):
   - Implemented all three functions with proper validation and error handling

4. **Interface** (`descope/sdk/mgmt.go`):
   - Updated `OutboundApplication` interface with the three new methods

5. **Mock Implementation** (`descope/tests/mocks/mgmt/managementmock.go`):
   - Added mock fields and methods for testing

6. **Unit Tests** (`descope/internal/mgmt/outbound_application_test.go`):
   - Comprehensive tests for all three functions
   - Success and error cases
   - Edge cases (e.g., deleting by appID only, userID only, or both)

All tests pass successfully, and the code is ready for use!

---
*Created by [Shuni](https://github.com/descope/shuni) 🐕*

### CODE REVIEW

File: descope/internal/mgmt/outbound_application.go [hunk 1]
```diff
@@ -113,6 +113,61 @@ func makeCreateUpdateOutboundApplicationRequest(app *descope.OutboundApp) map[st
...
+	if id == "" {
+		return utils.NewInvalidArgumentError("id")
+	}
+
+	req := &api.HTTPRequest{
+		QueryParams: map[string]string{"id": id},
+	}
+	_, err := s.client.DoDeleteRequest(ctx, api.Routes.ManagementOutboundApplicationDeleteTokenByID(), req, "")
+	return err
+}
```
[the remark this answers is not in the document]
Comment (omercnet) [reply descope/internal/mgmt/outbound_application.go]: ✅ Fixed! Updated the error handling to explicitly check for errors and return nil on success, matching the pattern in descoper.go and sso.go.

**Review by dorsha — APPROVED** [review]
